### ISSUE TITLE
[AVM Question/Feedback]: Output of `what-if` (if any) can't be trusted

### ISSUE THREAD

**Issue by o-l-a-v** [Azure/bicep-registry-modules#1366]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

AVM suffers the same problem as CARML: Bicep / ARM `what-if` doesn't handle nested modules, and thus shows little to no output. Which means it's spray and pray for deploying stuff with AVM, as `what-if` can't be trusted.

Is anything being done about this from AVM point of view?

Related issues:

* https://github.com/Azure/arm-template-whatif/issues/157

**Comment by neilmca-inc** [comment]
I don't think it matters where you get your modules from - the related issue you mention is pretty much where it is right now - awaiting a fix - but I don't think it's coming anytime soon to be a reliable yardstick

**Comment by o-l-a-v** [comment]
Sure, I agree. Yet doc doesn't mention a single issue about `what-if`:

* https://azure.github.io/Azure-Verified-Modules

So from AVM perspective this issue doesn't exist.

**Comment by AlexanderSehr** [comment]
Hey @o-l-a-v,
thank you for your issue. I responded to a very similar issue recently in this comment: https://github.com/Azure/bicep-registry-modules/issues/1292#issuecomment-2014612154

The long and short of it is that it is like @neilmca-inc said - it's not a module-specific issue (in this case AVM), but an overall flaw of the WhatIf functionality that should be raised here: https://github.com/Azure/arm-template-whatif (if not already existing).

It's unfortuantely not an issue we can do much about from our end as we cannot influence the language feature itself. And because of that I don't think it was considered to document the issue on our end either. That said, I of course agree with you, that it's quite a pitty that this is an issue in the first place and I hope the Product Group will get around to invest into it again soon.

cc: @jtracey93 @eriqua @ChrisSidebotham @matebarabas

**Comment by AlexanderSehr** [comment]
> Thanks @AlexanderSehr. AVM can however, and should in my opinion, inform users of AVM about this shortcoming. As in mention this limitation front and center in the documentation.
> 
> One of the main drivers for IaC is predictability by the ability to preview / plan / what-if. Be upfront about that shortcoming if choosing Bicep. Link to some issue people can vote on. And maybe suggest using AVM Terraform modules for now, if predictability is important.
> 
> One more thing: There already exists an issue in the repo you mentioned, I mentioned it in the issue description.

Hey @o-l-a-v I don't per se disagree - what-if IS very important and trust me I'd wish it would work better than it does too. I'm jut (in general) sceptical to documenting issues of external features that don't directly tie into a module's capabilties. In other words, it's on the 'solution'-side of the story, less than the CI/CD-side of this repository. 

Back in CARML we used to have a page for ['Solution creation'](https://github.com/Azure/ResourceModules/wiki/Solution%20creation) which would have been a perfect place to call this issue out. The only Wiki the BRM repository has is the [AVM Wiki](https://azure.github.io/Azure-Verified-Modules/) (which to my knowledge doesn't have a counterpart as of today). 

Regardless, if we were to call it out, where would you'd like to see this? In the AVM Wiki, here on the Public-Bicep-Registry CI/CD? Subjectively, I think the former would make more sense in the form of e.g., a  'Known issues' section. Please share your thoughts.

**Comment by o-l-a-v** [comment]
I like the idea of a "known issues/limitations" page on AVM wiki. Where issue inherited by using tools outside AVM are mentioned too, like ARM/Bicep what-if.

**Comment by AlexanderSehr** [comment]
Hey @o-l-a-v , thanks for the feedback. If you don't mind, I'll transfer the issue to the AVM repository where we'd work on the Wiki and can add the disclaimer.
